auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackIDPPrompt.java

```diff
@@ -186,7 +194,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             authResultTask
                     .addOnFailureListener(
                             new TaskFailureLogger(TAG, "Error linking with credential"))
-                    .addOnCompleteListener(new FinishListener());
+                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            FirebaseAuth auth = FirebaseAuth.getInstance();
+                            if (!task.isSuccessful() && task.getException() instanceof FirebaseAuthUserCollisionException) {
+                                setIntent(new Intent().putExtras(mActivityHelper.getMergeFailedIntent()));
+                                auth.signInWithCredential(mPrevCredential != null ? mPrevCredential : newCredential)
+                                        .addOnFailureListener(
+                                                new TaskFailureLogger(
+                                                        TAG, "Error linking with credential"))
+                                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                                            @Override
+                                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                                if (task.isSuccessful()) {
+                                                    mActivityHelper.dismissDialog();
+                                                    finish(Activity.RESULT_OK, getIntent());
```

Comment:
I'm actually not sure what's going on here. It looks like if we fail to link with a credential we try to sign in anyway.  If the sign in succeeds (but the linking did not) we say it was a success?